### DISCUSSION TITLE
In random mode player don't play all the album or list

### DISCUSSION THREAD

**Discussion by yopmismo** [rern/rAudio#236] [Fixed Issues]
Hello,
If I set random mode in a list or album, player start playing from a random song, but when stop it left behind the tracks before the star song and play these songs.
If you turn on "random" and "consume mode" you can see easily the tracks the payer didn't play after random mode stop.

**Comment by rern** [comment]
`random` - Play tracks from Playlist in random order.
`consume` - Remove each track from Playlist after played.

Describe your issue which one didn't function properly.

**Reply by yopmismo** [reply]
Random, if I use random mode some tracks are left without play.

**Reply by rern** [reply]
I think MPD implements `random` that literally just random some and plays. Another mode that should meet your need is `shuffle` playlist, the icon next to `Roll the dice`.

**Reply by yopmismo** [reply]
I have a Bryston BDP-2 running MPD and it haven't this issue, but seems a lot of work to do.
The same with the 32 bit showed instead of 24 bits for tracks that was addressed on BDP-2.
Thank you for your time...

**Reply by voterpublic** [reply]
Firstly, thanks @rern for a brilliant continuation of Rune :).
I agree with what @yopmismo has brought up but i would not classify it as a bug. 
It is just not "random".
If you activate "roll of the dice" (using it to, in effect, create a random continuous playlist from your library) with an hour you begin to listen to the same artist songs (same album too) again & again.
It will mostly choose the same "random" songs, the same "random" albums, the same "random" artist everytime "roll of the dice" is used. It does not cycle through the library at all, many artists & albums are never played (I have "roll of the dice" selected up to 12hrs at a time).
This occurs regardless if consume mode is active or not.
Is there an easy fix?
Thanks

**Comment by rern** [comment]
Improved with 20220902 update. Now it keeps tracks log to select only tracks not already played.

**Reply by voterpublic** [reply]
Too cool @rern , many thanks
